prow/plugins/invalidcommitmsg/invalidcommitmsg.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package invalidcommitmsg adds the "do-not-merge/invalid-commit-message"
+// label on PRs containing commit messages with @mentions or
+// keywords that can automatically close issues.
+package invalidcommitmsg
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+const (
+	pluginName            = "invalidcommitmsg"
+	invalidCommitMsgLabel = "do-not-merge/invalid-commit-message"
+	commentBody           = `[Keywords](https://help.github.com/articles/closing-issues-using-keywords) which can automatically close issues and at(@) mentions are not allowed in commit messages.
+
+Please remove these keywords from the following commit messages: `
+)
+
+var (
+	closeIssueRegex = regexp.MustCompile(`(([cC]los(?:e[sd]?))|([fF]ix(?:(es|ed)?))|([rR]esolv(?:e[sd]?)))[\s:]+(\w+/\w+)?#(\d+)`)
+	atMentionRegex  = regexp.MustCompile(`@[-\w]+`)
```

Comment:
this will false positive email addresses, right? Probably want to require a non word character or start-of-string in front of the @

Reply:
I was wondering about this. How does this affect collaborative commits, made by multiple people?

Reply:
and let's add a test case containing an email address

Reply:
Done. Changed the regex and added a test.